File: python-backend/src/models/frage.py
class Frage:
    """Repräsentiert eine Quiz-Frage"""
    
    def __init__(self, id, text, category, difficulty, 
                 answered_correctly=0, answered_incorrectly=0):
        self.id = id
        self.text = text
        self.category = category
        self.difficulty = difficulty
        self.answered_correctly = answered_correctly
        self.answered_incorrectly = answered_incorrectly
        self.answers = []
    
    def add_answer(self, answer):
        """Fügt eine Antwort zur Frage hinzu"""
        self.answers.append(answer)
    
    def to_dict(self):
        return {
            'id': self.id,
            'text': self.text,
            'category': self.category,
            'difficulty': self.difficulty,
            'answered_correctly': self.answered_correctly,
            'answered_incorrectly': self.answered_incorrectly,
            'answers': [a.to_dict() for a in self.answers]
        }

File: Android/app/src/main/java/com/quizly/android/data/model/Frage.kt
package com.quizly.android.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Ported from Frage.cs / frage.py. The live API may return category/difficulty as
 * nested objects or as plain foreign-key ids - kept nullable/best-effort either way,
 * the lenient JSON config (ignoreUnknownKeys/coerceInputValues) means a mismatch here
 * just leaves these fields null rather than crashing the whole response.
 */
@Serializable
data class Frage(
    val id: Int,
    val text: String,
    @SerialName("category")
    val categoryId: Int? = null,
    @SerialName("difficulty")
    val difficultyLevel: Int? = null,
    @SerialName("answered_correctly")
    val answeredCorrectly: Int = 0,
    @SerialName("answered_incorrectly")
    val answeredIncorrectly: Int = 0,
    val answers: List<Antwort> = emptyList()
) {
    fun isAnswerCorrect(answerId: Int): Boolean = answers.any { it.id == answerId && it.isCorrect }
}
